tests/rados/test_node_drain_customer_bug.py

```diff
@@ -108,37 +111,47 @@ def run(ceph_cluster, **kw):
-            break
-    # Select the node with OSD nodes weight/re-weights are 0 if none of the hosts have the _no_schedule label.
-    if drain_host is None:
-        osd_down_node_list = get_zero_osd_weight_list(rados_obj)
-        if len(osd_down_node_list) == 0:
-            log.info("None of the OSD weight and reweight are 0")
-        else:
-            drain_host, _ = list(osd_down_node_list.items())[0]
-            log.info(f"The drain operation is performing on {drain_host} node")
-    # Pick any random OSD node if none of hosts have the _no_schedule label and all OSDs are up.
-    if drain_host is None:
-        osd_hosts = rados_obj.get_osd_hosts()
-        log.info(f"The osd nodes list are-{osd_hosts}")
-        drain_host = random.choice(osd_hosts)
-
-    init_time, _ = installer.exec_command(cmd="sudo date '+%Y-%m-%d %H:%M:%S'")
-    log.info(f"The test execution  started at - {init_time} ")
+
     try:
+        # Check any nodes that has _no_schedule label
+        for node in out:
+            if "_no_schedule" in node["labels"]:
+                drain_host = node["hostname"]
+                log.info(f"The {drain_host} is identified to drain from the cluster")
+                break
+        # Select the node with OSD nodes weight/re-weights are 0 if none of the hosts have the _no_schedule label.
+        if drain_host is None:
+            osd_down_node_list = get_zero_osd_weight_list(rados_obj)
+            if len(osd_down_node_list) == 0:
+                log.info("None of the OSD weight and reweight are 0")
+            else:
+                drain_host, _ = list(osd_down_node_list.items())[0]
+                log.info(f"The drain operation is performing on {drain_host} node")
+        # Pick any random OSD node if none of hosts have the _no_schedule label and all OSDs are up.
+        if drain_host is None:
+            osd_hosts = rados_obj.get_osd_hosts()
+            log.info(f"The osd nodes list are-{osd_hosts}")
+            drain_host = random.choice(osd_hosts)
+
+        init_time, _ = installer.exec_command(cmd="sudo date '+%Y-%m-%d %H:%M:%S'")
+        log.info(f"The test execution  started at - {init_time} ")
         osd_count_before_test = get_node_osd_list(rados_obj, ceph_nodes, drain_host)
         log.info(
             f"The OSDs in the drain node before starting the test- {osd_count_before_test} "
         )
         rados_obj.set_service_managed_type("osd", unmanaged=True)
+        # Re-producing the bug by running  mgr start and removing OSD parallely
         time.sleep(10)
         mgr_daemon.start()
-        service_obj.remove_custom_host(host_node_name=drain_host)
+        if bug_exists:
+            log.info("Reproducing  the bug")
+            rm_host = utils.get_node_by_id(ceph_cluster, drain_host)
+            log.info(
+                f"Identified host : {rm_host.hostname} to be removed from the cluster"
+            )
+            host_drain(cephadm, remove_osd_node=rm_host.hostname)
```

Comment:
We are not checking the result of the method. I feel that we can check the result of repro here as well.